src/store/data/actions/loadExpressions.ts

```diff
@@ -78,28 +79,31 @@ export default async function ({
     };
 
     //Listen for message callback saying we got some links
-    linksWorker.addEventListener("message", (e) => {
+    linksWorker.addEventListener("message", async (e) => {
       const linkQuery = e.data.perspectiveQueryLinks;
 
-      for (const link of linkQuery) {
-        //Hash the link data as the key for map and check if it exists in the store
+      const links = linkQuery.filter((link: any) => {
         const currentExpressionLink =
-          channel.currentExpressionLinks[link.data.target];
+        channel.currentExpressionLinks[link.data.target];
         const currentExpression =
-          channel.currentExpressionMessages[link.data.target];
-
-        if (!currentExpressionLink || !currentExpression) {
-          //Run expression worker to try and get expression on link target
-          expressionWorker.postMessage({
-            retry: 50,
-            interval: 5000,
-            query: print(GET_EXPRESSION),
-            variables: { url: link.data.target },
-            callbackData: { link },
-            name: `Get expression data from channel links ${channel.name}`,
-            dataKey: "expression",
-          });
-        }
+        channel.currentExpressionMessages[link.data.target];
+
+        return !currentExpressionLink || !currentExpression;
+      });
+
+      const linksHash = links.map((link: any) => link.data.target);
```

Comment:
Will this map fail if the above `linkQuery.filter()` returns an `ExpressionRef` and not a `LinkExpression`?

Reply:
Can the linkworker return an ExpressionRef instead of LinkExpression?

Reply:
lol code reading error on my side, looks good 👍

Reply:
Ok cool 😅